packages/algorand-js/src/errors/errors-list.ts

```diff
@@ -116,23 +116,47 @@ by an index that does not exist.`
     title: "Input is outside the valid uint8 range of 0 to 255",
     description: `Input is outside the valid uint8 range of 0 to 255`
   },
-  ASSERT_FIELD_LENGTH: {
+  ASSERT_LENGTH: {
     number: 16,
     message: "Invalid Field Length Expected: '%exp%' Got: '%got%', Line : %line% ",
     title: PARSE_ERROR,
-    description: ``
+    description: `Expected: '%exp%' Got: '%got%`
   },
   INVALID_ADDR: {
     number: 17,
     message: "Invalid Address '%addr%', Line: %line%",
     title: PARSE_ERROR,
-    description: ``
+    description: `Invalid Address '%addr%`
   },
   PRAGMA_VERSION_ERROR: {
     number: 18,
     message: "Pragma version Error - Expected: version, got: %got%, Line: %line%",
     title: PARSE_ERROR,
     description: ``
+  },
+  INVALID_BASE64: {
+    number: 19,
+    message: "Invalid Base64 Error - value %val% is not base64, Line: %line%",
+    title: PARSE_ERROR,
+    description: `value %exp% is not base64`
+  },
+  INVALID_BASE32: {
+    number: 20,
+    message: "Invalid Base32 Error - value %val% is not base32, Line: %line%",
+    title: PARSE_ERROR,
+    description: `value %exp% is not base32`
+  },
+  DECODE_ERROR: {
+    number: 21,
+    message: "Invalid Decode Data - value %val% is invalid, Line: %line%",
+    title: "Decode Error",
+    description: `value %exp%`
+  },
+  UNKOWN_DECODE_TYPE: {
+    number: 22,
+    message: "Invalid Decode Type - value %val% is unknown, Line: %line%",
+    title: "Unkown Decode Type",
+    description: `value %exp% is unknown`
```

Comment:
Since we have quite a few parser errors now as well, do you think we should separate them from execution errors? this list seems too long. We can have 
+ execution error: stack related, logic rejection etc.. 
+ parsing errors: invalid string, encode/decode etc..
Just a thought. I think they would be more organized this way. Let me know what you think

Reply:
Good idea

Reply:
but i don't think it will be more organized as currently we have `tealErrors`, most of them are overlapping, only 2-3 errors will be moved to Parsing error. some are used in constructor also

Reply:
Okay, if only 2-3 errors will be moved then it makes sense to leave as it is.